ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py

```diff
@@ -136,3 +147,173 @@ def wan_local_path_groups(self: SharedUtils) -> list:
             local_path_groups_dict[path_group_name]["interfaces"].extend(carrier["interfaces"])
 
         return list(local_path_groups_dict.values())
+
+    @cached_property
+    def this_wan_route_server(self: SharedUtils) -> dict:
+        """
+        Returns the instance for this wan_rs found under wan_route_servers.
+        Should only be called when the device is actually a wan_rs. Will raise if not found.
+        """
+        wan_route_servers = get(self.hostvars, "wan_route_servers", default=[])
+        return get_item(wan_route_servers, "hostname", self.hostname, default={})
+
+    def get_public_ip_for_wan_interface(self: SharedUtils, interface: dict) -> str:
+        """
+        Takes a dict which looks like `l3_interface` from node config
+
+        Return the `ip` field unless for WAN route-servers with interface ip 'dhcp',
+        where we try to find the IP under wan_route_servers.path_groups.interfaces.
+        Raises if the IP is not found.
+        """
+        if interface["ip"] != "dhcp" or self.wan_role != "server":
+            return interface["ip"]
+
+        for path_group in self.this_wan_route_server.get("path_groups", []):
+            if (found_interface := get_item(path_group["interfaces"], "name", interface["name"])) is None:
+                continue
+
+            if found_interface.get("ip_address") is not None:
+                return found_interface["ip_address"]
```

Comment:
the ip address for a route server defined under wan_route_servers.path_groups.interfaces should take precedence

Reply:
done